src/pykeen/losses.py

```diff
@@ -637,7 +637,7 @@ def process_slcwa_scores(
             # negative_scores have already been filtered in the sampler!
             # (dense) softmax requires unfiltered scores / masking
             negative_scores_ = torch.zeros_like(batch_filter, dtype=positive_scores.dtype)
-            negative_scores_[batch_filter] = negative_scores
+            negative_scores_[batch_filter] = negative_scores.view(1, -1)
```

Comment:
what is the shape of `negative_scores` in the failing test case?

Reply:
Its [164] and it’s supposed to be [164,1]. It complained it couldn’t broadcast

Reply:
I'll check this again tomorrow. From my understanding, the LHS should be 1d (since we apply a boolean mask of the same shape as the `negative_scores_` tensor. The RHS should also be 1d, since it also originates from applying the same boolean mask to a score tensor, which *should* be of shape `(batch_size, num_negatives)` (i.e., exactly the size of the boolean mask)

Reply:
Okay. I’ll revert this locally and get you the direct trace back